sds_data_manager/lambda_code/SDSCode/dependency_config.py

```diff
@@ -1,58 +1,505 @@
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="lo-pha",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="lo-pha",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="hi-pha",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="hi-pha",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="hi-omni",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="hi-omni",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="hi-sectored",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="hi-sectored",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="lo-ialirt",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="lo-ialirt",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    PreProcessingDependency(
+        primary_instrument="codice",
+        primary_data_level="l1a",
+        primary_descriptor="hi-ialirt",
+        dependent_instrument="codice",
+        dependent_data_level="l1b",
+        dependent_descriptor="hi-ialirt",
+        relationship="HARD",
+        direction="DOWNSTREAM",
+        reverse_direction=True,
+    ),
+    # <---- GLOWS Dependencies ---->
+    # TODO: add GLOWS l0 to l1a dependencies
+    # <---- HI Dependencies ---->
+    PreProcessingDependency(
+        primary_instrument="hi",
         primary_data_level="l0",
         primary_descriptor="raw",
-        dependent_instrument="swapi",
-        dependent_data_level="l1",
+        dependent_instrument="hi",
+        dependent_data_level="l1a",
         dependent_descriptor="all",
         relationship="HARD",
         direction="DOWNSTREAM",
+        reverse_direction=True,
     ),
     PreProcessingDependency(
-        primary_instrument="swe",
+        primary_instrument="hi",
+        primary_data_level="l1a",
+        primary_descriptor="45sensor-histogram",
```

Comment:
I assume that the `primary_descriptor` must match filename descriptors. Right now, this string is hard coded here and in the `imap_processing` repo global attributes definition for the file and must match. It seems like we need to decide on a single source of truth for where these strings are defined. This is not for this PR, but a ticket should be written to capture this.

Reply:
I totally agree. I have been looking at CDF attrs' `logical_source` to match these but would be nice to have source of truth for descriptor as well. I wonder if we can use imap-data-access. But I will create a ticket to capture this.